packages/core/stellar/src/non-fungible.ts

```diff
@@ -0,0 +1,395 @@
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.approve);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.approve_for_all);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.get_approved);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.is_approved_for_all);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.name);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.symbol);
+  c.addTraitFunction(nonFungibleTokenTrait, baseFunctions.token_uri);
+
+  if (pausable) {
+    c.addUseClause('stellar_pausable_macros', 'when_not_paused');
+    c.addFunctionTag(baseFunctions.transfer, 'when_not_paused', nonFungibleTokenTrait);
+    c.addFunctionTag(baseFunctions.transfer_from, 'when_not_paused', nonFungibleTokenTrait);
+  }
+}
+
+function addBurnable(c: ContractBuilder, pausable: boolean) {
+  c.addUseClause('stellar_non_fungible', 'burnable::NonFungibleBurnable');
+  c.addUseClause('soroban_sdk', 'Address');
+
+  const nonFungibleBurnableTrait = {
+    traitName: 'NonFungibleBurnable',
+    structName: c.name,
+    tags: ['contractimpl'],
+    section: 'Extensions',
+  };
+
+  c.addTraitFunction(nonFungibleBurnableTrait, burnableFunctions.burn);
+  c.addTraitFunction(nonFungibleBurnableTrait, burnableFunctions.burn_from);
+
+  if (pausable) {
+    c.addUseClause('stellar_pausable_macros', 'when_not_paused');
+    c.addFunctionTag(burnableFunctions.burn, 'when_not_paused', nonFungibleBurnableTrait);
+    c.addFunctionTag(burnableFunctions.burn_from, 'when_not_paused', nonFungibleBurnableTrait);
+  }
+}
+
+function addEnumerable(c: ContractBuilder, burnable: boolean) {
+  c.addUseClause('stellar_non_fungible', 'enumerable::{NonFungibleEnumerable, Enumerable}');
+  //c.addUseClause('stellar_default_impl_macro', 'default_impl');
+
+  const nonFungibleEnumerableTrait = {
+    traitName: 'NonFungibleEnumerable',
+    structName: c.name,
+    tags: ['defaultimpl', 'contractimpl'],
+    section: 'Extensions',
+  };
+  c.addTraitImplBlock(nonFungibleEnumerableTrait);
+
+  c.overrideAssocType('NonFungibleToken', 'type ContractType = Enumerable;');
+
+  if (burnable) {
+    c.overrideAssocType('NonFungibleBurnable', 'type ContractType = Enumerable;');
+  }
+
+  // Below is not required due to `defaultimpl` macro. If we require to customize the functions,
+  // then we should:
+  // 1. get rid of the `defaultimpl` macro form `tags` above,
+  // 2. get rid of `c.addImplementedTrait(nonFungibleEnumerableTrait);` line,
+  // 3. uncomment the section below:
+  // 4. uncomment `get_owner_token_id` and `get_token_id` from `enumerableFunctions` definition
+  /*
+  c.addFunction(nonFungibleEnumerableTrait, enumerableFunctions.total_supply);
+  c.addFunction(nonFungibleEnumerableTrait, enumerableFunctions.get_owner_token_id);
+  c.addFunction(nonFungibleEnumerableTrait, enumerableFunctions.get_token_id);
+  */
+}
+
+function addConsecutive(c: ContractBuilder, burnable: boolean, access: Access) {
```

Comment:
Looks like burnable is no longer used here
```suggestion
function addConsecutive(c: ContractBuilder, access: Access) {
```